access/msteams/plugindata.go

```diff
@@ -26,14 +27,22 @@ func DecodePluginData(dataMap map[string]string) PluginData {
 
 	data.AccessRequestData = plugindata.DecodeAccessRequestData(dataMap)
 
-	if channelID, timestamp := dataMap["channel_id"], dataMap["timestamp"]; channelID != "" && timestamp != "" {
-		data.TeamsData = append(data.TeamsData, TeamsMessage{ID: channelID, Timestamp: timestamp})
-	}
 	if str := dataMap["messages"]; str != "" {
 		for _, encodedMsg := range strings.Split(str, ",") {
-			parts := strings.Split(encodedMsg, "/")
-			if len(parts) == 3 {
-				data.TeamsData = append(data.TeamsData, TeamsMessage{ID: parts[0], Timestamp: parts[1], RecipientID: parts[2]})
+			decodedMsg, err := base64.StdEncoding.DecodeString(encodedMsg)
+			if err != nil {
+				// Backward compatibility
+				parts := strings.Split(encodedMsg, "/")
+				if len(parts) == 3 {
+					data.TeamsData = append(data.TeamsData, TeamsMessage{ID: parts[0], Timestamp: parts[1], RecipientID: parts[2]})
+				}
```

Comment:
We should remove this path later on right?

Can we add a TODO here? Maybe to remove this when we release v12 ?